dialogy/plugins/text/classification/xlmr.py

```diff
@@ -74,61 +66,94 @@ def __init__(
         self.use_prompt = use_prompt
         self.null_prompt_token = null_prompt_token
         self.prompts_map = prompts_map
+        self.threshold = threshold
+        self.skip_labels = set(skip_labels or set())
+        self.threshold = threshold
+        self.round = score_round_off
 
         self.purpose = kwargs.pop("purpose", const.TRAIN)
-        self.use_cuda = self.purpose == const.TRAIN
 
         if args_map and self.purpose not in args_map:
             raise ValueError(
-                f"Attempting to set invalid {args_map}. "
-                f"It is missing {self.purpose}. {self.purpose} has to be one of "
+                f"Attempting to set invalid `args_map`. "
+                f"It is missing {self.purpose}. `purpose` has to be one of "
                 f"{const.TRAIN}, {const.TEST}, {const.PRODUCTION} in configs."
             )
         self.args_map = args_map[self.purpose]
 
+        self.use_calibration = self.args_map.get(const.MODEL_CALIBRATION, False)
         self.model_dir = self.args_map.get("best_model_dir")
-
         if not self.model_dir:
             raise ValueError(
                 f"'best_model_dir' missing in passed args_map."
             )
-
-        self.use_calibration = self.args_map.get(const.MODEL_CALIBRATION, False)
-
-        self.labelencoder_file_path = os.path.join(
-            self.model_dir, const.LABELENCODER_FILE
-        )
+        
         self.ts_parameter: float = (
             read_from_json(
                 [const.TS_PARAMETER], self.model_dir, const.CALIBRATION_CONFIG_FILE
             ).get(const.TS_PARAMETER)
             or 1.0
         )
 
-        self.threshold = threshold
-        self.skip_labels = set(skip_labels or set())
-
-        self.round = score_round_off
+        # flag that specifies whether plugin is being imported externally solely for model
+        imported = kwargs.get("imported", False)
 
-        self.kwargs = kwargs or {}
+        if self.purpose == const.TRAIN or imported:
+            self.use_cuda = self.purpose == const.TRAIN
```

Comment:
Lets set `self.use_cuda` to True only when cuda is actually available. [This is how](https://github.com/skit-ai/dialogy/commit/0472e138322b877e3793871cc199020b6ce4f9ee) we do right now on `0.9.x`. Otherwise this breaks training on CPU which is sometimes needed just to do sanity testing of training workflow.